executor/load_data.go

```diff
@@ -358,14 +392,48 @@ func (e *LoadDataWorker) Load(
 		importer.LogicalImportMode,
 		e.Ctx.GetSessionVars().User.String(),
 	)
+	if err != nil {
+		return 0, err
+	}
+
+	if e.controller.Detached {
+		go func() {
+			detachedCtx := context.Background()
+			detachedCtx = kv.WithInternalSourceType(detachedCtx, kv.InternalLoadData)
+			// error is stored in system table, so we can ignore it here
+			//nolint: errcheck
+			_ = e.doLoad(detachedCtx, readerInfos, jobID)
+		}()
+		return jobID, nil
+	}
+	return jobID, e.doLoad(ctx, readerInfos, jobID)
+}
+
+func (e *LoadDataWorker) doLoad(
+	ctx context.Context,
+	readerInfos []LoadDataReaderInfo,
+	jobID int64,
+) (err error) {
+	defer func() {
+		if e.ownSession {
+			e.putSysSessionFn(ctx, e.Ctx)
```

Comment:
maybe remove `e.Ctx`, we're using `e.ctx` and `e.Ctx` mixingly, it's same as ctx in `InsertValues`

Reply:
it's made public for LOAD DATA LOCAL where the clientConn.handleLoadData check Ctx.GetSessionVars().Killed to handle reading packet. Maybe I can unify the CANCEL LOAD DATA JOB with this checking killed in next PR.